src/router.js

```diff
@@ -144,10 +132,21 @@ const router = new Router({
  * if so, then ensure that they are correctly logged in as a valid user
  * If not logged in, prevent all access and redirect them to login page
  * */
-router.beforeEach((to, from, next) => {
+router.beforeEach(async (to, from, next) => {
   progress.start();
+  // Confirm that the config is initialized, if not wait for it to be
+  if (!configFetched) {
+    await $store.dispatch(Keys.WAIT_FOR_CONFIG);
+    getConfig();
+  }
   if (to.name !== 'login' && !isAuthenticated()) next({ name: 'login' });
-  else next();
+  // If routing to home, redirect to the correct view
+  else if (to.path === '/') {
+    if (from.name === defaultView) {
+      next(false);
+      progress.end();
+    } else next({ name: defaultView });
+  } else next();
```

Comment:
There seems to be some new issues with the router. Navigating to pages works fine, but sometimes if you visit a sub-page directly by URL then you get an error.

For example, https://deploy-preview-549--dashy-dev.netlify.app/home/getting-started Vs previous version: https://dev.dashy.to/workspace

```
Uncaught TypeError: Cannot read property 'routingMode' of undefined
```